pkg/odo/cli/component/push.go

```diff
@@ -176,7 +175,7 @@ func (po *PushOptions) Validate() (err error) {
 	// If the experimental flag is set and devfile is present, then we do *not* validate
 	// TODO: We need to clean this section up a bit.. We should also validate Devfile here
 	// too.
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(po.DevfilePath) {
```

Comment:
I'm wondering if we need an additional check here for when the pushtarget is set to Docker, but experimental mode isn't set, and so we error out? Since we're keeping Docker support as experimental.

Some commands, like `odo url create` require both to be set, but others like `odo push` and `odo delete` don't.